### PR TITLE
Add option to fail the build

### PR DESCRIPTION
**Pull request by thorncp**
Adds a configuration option to set the PR status to 'error' when
violations are found.

*.hound.yml*
```yaml
fail_on_violations: true
```

more_dumb_stuff_with_custom_config_inheriting_by_thorncp_ _pull_request__59_ _thornco_dumb-ruby

https://trello.com/c/MtnDjmfF

### CODE REVIEW

File: app/models/commit_status.rb [hunk 1]
```diff
@@ -14,7 +14,12 @@ def set_success(violation_count)
     github.create_success_status(repo_name, sha, message)
   end
 
-  def set_failure
+  def set_failure(violation_count)
+    message = I18n.t(:success_status, count: violation_count)
```
Comment (gylaz) [inline app/models/commit_status.rb]: Should we rename `success_status` to something more general?

**Comment by thorncp** [comment]
I just realized this doesn't check the violation count, so it'll always set the error status if it's enabled.

**Comment by salbertson** [comment]
Why are we making this an option?

**Comment by thorncp** [comment]
People have asked for it. I think it'd be an interesting experiment to try, even if it doesn't stick around.

https://trello.com/c/MtnDjmfF

**Comment by thorncp** [comment]
Updated to not fail with 0 violations in fb0c4836a3567f74718d093c78a9b695fc8c2a0d.

File: app/models/commit_status.rb (hunk 1 above)
Comment (thorncp) [inline app/models/commit_status.rb]: Changed to `complete_status` in 917d100409fe3514d7dd13d03b873d3e9d1415b9

**Comment by thorncp** [comment]
Enabled this on the Hound main repo in 87543b7810d80e389fef0752c411c0f2f425f258

**Pull request merged**